phoenix-core/src/test/java/org/apache/phoenix/index/PrepareIndexMutationsForRebuildTest.java

```diff
@@ -0,0 +1,626 @@
+        byte[] idxKeyBytes2 = com.google.common.primitives.Bytes.toArray(idxKey2);
+        Put idxPut2 = new Put(idxKeyBytes2);
+        addCellToPutMutation(idxPut2,
+                QueryConstants.DEFAULT_COLUMN_FAMILY_BYTES,
+                Bytes.toBytes("0:C2"),
+                2,
+                Bytes.toBytes("v2"));
+        addEmptyColumnToIndexPutMutation(idxPut2, info.indexMaintainer, 2);
+        expectedIndexMutations.add(idxPut2);
+
+        Delete idxDel = new Delete(idxKeyBytes);
+        addCellToDelMutation(idxDel,
+                QueryConstants.DEFAULT_COLUMN_FAMILY_BYTES,
+                null,
+                2,
+                KeyValue.Type.DeleteFamily);
+        expectedIndexMutations.add(idxDel);
+
+        assertEqualMutationList(expectedIndexMutations, actualIndexMutations);
+    }
+
+    @Test
+    public void testForMultipleFamilies() throws Exception {
+        SetupInfo info = setup(TABLE_NAME,
+                INDEX_NAME,
+                "ROW_KEY VARCHAR, CF1.C1 VARCHAR, CF2.C2 VARCHAR",
+                "CF1.C1",
+                "ROW_KEY",
+                "CF2.C2");
+
+        Put dataPut = new Put(Bytes.toBytes(ROW_KEY));
+        addCellToPutMutation(dataPut,
+                Bytes.toBytes("CF1"),
+                Bytes.toBytes("C1"),
+                1,
+                Bytes.toBytes("v1"));
+        addCellToPutMutation(dataPut,
+                Bytes.toBytes("CF2"),
+                Bytes.toBytes("C2"),
+                1,
+                Bytes.toBytes("v2"));
+        addEmptyColumnToDataPutMutation(dataPut, info.pDataTable, 1);
+
+        Delete dataDel = new Delete(Bytes.toBytes(ROW_KEY));
+        addCellToDelMutation(dataDel,
+                Bytes.toBytes("CF1"),
+                Bytes.toBytes("C1"),
+                2,
+                KeyValue.Type.DeleteColumn);
+
+        List<Mutation> actualIndexMutations = IndexRebuildRegionScanner.prepareIndexMutationsForRebuild(info.indexMaintainer,
+                dataPut,
+                dataDel);
+
+        List<Mutation> expectedIndexMutation = new ArrayList<>();
+
+        byte[] idxKeyBytes = generateIndexRowKey("v1");
+
+        Put idxPut1 = new Put(idxKeyBytes);
+        addCellToPutMutation(idxPut1,
+                Bytes.toBytes("CF2"),
+                Bytes.toBytes("CF2:C2"),
+                1,
+                Bytes.toBytes("v2"));
+        addEmptyColumnToIndexPutMutation(idxPut1, info.indexMaintainer, 1);
+        expectedIndexMutation.add(idxPut1);
+
+        Put idxPut2 = new Put(generateIndexRowKey(null));
+        addCellToPutMutation(idxPut2,
+                Bytes.toBytes("CF2"),
+                Bytes.toBytes("CF2:C2"),
+                2,
+                Bytes.toBytes("v2"));
+        addEmptyColumnToIndexPutMutation(idxPut2, info.indexMaintainer, 2);
+        expectedIndexMutation.add(idxPut2);
+
+        Delete idxDel = new Delete(idxKeyBytes);
+        addCellToDelMutation(idxDel,
```

Comment:
Comment to explain that the DeleteColumn on CF1 removes the only indexed column and thus removes the covered column in CF2

Reply:
The Explaination will be the same as tesDelOnIndexColumn. I will add them there as well.